src/Language/PureScript/Errors.hs

```diff
@@ -724,7 +724,7 @@ prettyPrintSingleError (PPEOptions codeColor full level showWiki) e = flip evalS
     renderSimpleErrorMessage (InvalidInstanceHead ty) =
       paras [ line "Type class instance head is invalid due to use of type"
             , markCodeBox $ indent $ typeAsBox ty
-            , line "All types appearing in instance declarations must be of the form T a_1 .. a_n, where each type a_i is of the same form."
+            , line "All types appearing in instance declarations must be of the form T a_1 .. a_n, where each type a_i is of the same form. Unless the type is fully determined by other type class arguments via functional dependencies."
```

Comment:
Wording nitpick: I'd use a comma or parens here rather than a period. Although I wonder if we should just link to the wiki for further details here.